src/main/java/frc/robot/utils/InterpolateDouble.java

```diff
@@ -0,0 +1,32 @@
+package frc.robot.utils;
+
+import java.util.HashMap;
+
+public class InterpolateDouble {
+    private HashMap<Double, Double> map;
+
+    public InterpolateDouble(HashMap<Double, Double> map) {
+        this.map = map;
+    }
+
+    public double getValue(double key) {
+        if (map.containsKey(key)) {
+            return map.get(key);
+        }
+
+        double lowerKey = 0;
+        double upperKey = 0;
+        for (double k : map.keySet()) {
+            if (k < key && k > lowerKey) {
+                lowerKey = k;
+            } else if (k > key && k < upperKey) {
+                upperKey = k;
+            }
+        }
```

Comment:
Could we just require that the map keys are monotonically increasing? Then we would just have a for loop and once we find the lower key then the next key is the upper key and we don't need to keep searching